File: py/aoc2015/day1.py
SAMPLE_INPUT = [
        '(',
        ')',
        ')',
]

FILENAME = "../inputs/2015/day1.txt"

def _parse(filename):
    with open(filename) as f:
        data = f.read()
    return data

def part1():
    floor = 0
    for item in _parse(FILENAME):
        if item == '(':
            floor += 1
        elif item == ')':
            floor -= 1
    return floor


def part2():
    position = 0
    floor = 0
    for item in _parse(FILENAME):
        if item == '(':
            floor += 1
        elif item == ')':
            floor -= 1

        if floor == -1:
            break
        position += 1
    return position

if __name__ == "__main__":
    p1 = part1()
    print(p1)
    p2 = part2()
    print(p2)

File: elixir/aoc2015/lib/day1.ex
defmodule Day1 do
  @file_name "../../../inputs/2015/day1.txt"
  def read_file(file_name \\ @file_name) do
    {_, content} = File.read(file_name)
    String.trim(content)
  end

  def move_santa("(" <> rest, floor, character) do
    move_santa(rest, floor + 1, character + 1)
  end

  def move_santa(")" <> rest, floor, character) do
    case floor do
      -1 -> 
        IO.puts("PART2 solution: #{character}") 
        move_santa(rest, floor - 1, character + 1)
      _ -> move_santa(rest, floor - 1, character + 1)
    end
  end
  def move_santa("" <> _rest, floor, _character) do
    IO.puts("PART1 solution: #{floor} ")
  end
end

Day1.read_file
|> Day1.move_santa(0, 0)


